source/common/protobuf/message_validator_impl.cc

```diff
@@ -24,20 +24,31 @@ void WarningValidationVisitorImpl::onUnknownField(absl::string_view description)
   if (!it.second) {
     return;
   }
+
   // It's a new field, log and bump stat.
-  ENVOY_LOG(warn, "Unknown field: {}", description);
-  if (counter_ == nullptr) {
-    ++prestats_count_;
+  ENVOY_LOG(warn, "Unexpected field: {}", description);
+  if (unknown_counter_ == nullptr) {
+    ++prestats_unknown_count_;
   } else {
-    counter_->inc();
+    unknown_counter_->inc();
   }
 }
 
+void WarningValidationVisitorImpl::onDeprecatedField(absl::string_view description,
+                                                     bool soft_deprecation) {
+  onDeprecatedFieldDefault(description, soft_deprecation);
```

Comment:
Maybe just call it `onDeprecatedFieldCommon` or something at this point (since it's no longer inheritance).